tests/TestCase/Database/Schema/PostgresSchemaTest.php

```diff
@@ -493,11 +515,11 @@ public function testDescribeTableWithDefaults()
                 'autoIncrement' => null,
             ],
             'created' => [
-                'type' => 'timestamp',
+                'type' => 'timestampfractional',
```

Comment:
Changing the default reflected type could be problematic for applications. Is there a way we could make this feature opt-in?

Reply:
Wouldn't those using fractional datetime/timestamp field actually want they fractional part?

Reply:
I tried to work on this, but I'm not sure the best way. You can certainly `opt-out` by remapping `timestampfractional` to `DateTimeType`.

When converting SQL to Schema, postgres doesn't show the difference between TIMESTAMP and TIMESTAMP(6).  They both have the same type name and precision value.

Reply:
> Wouldn't those using fractional datetime/timestamp field actually want they fractional part?

I believe he is talking about legacy projects that upgrade to 4.0 where using postgresl + TIMESTAMP they have broken code that accidentally sets the microseconds.

With this, they would get the microseconds back out which could cause ... I don't know a parsing error?  Maybe a display error.  It would probably be minor unless it causes an overflow of some kind in a validation suite.

Reply:
I suppose another bug it could cause is SQL logic that looks for rows with the same timestamp. If their logic is accidentally setting microseconds in the I18n\Time instance that we are accidentally scrubbing, that could be a subtle error in future inserts.

Reply:
My concern was about people having conditional logic on abstract schema types. Previously they would have gotten `datetime` or `timestamp`, now they'll get something different. It could impact plugins like Bake too.

Reply:
@markstory I ran bake with datetime(6) and timestamp(6) with this change without any issue.  I'm not sure that anything changed really.  Is there something specific you'd like tested?

Reply:
That is what I was hoping to have tested. The FormHelper test you added is great :clap:

Reply:
I think FormHelperTest is missing a test where it builds a ->control() based on the field name. That should look up the default mapping for types.  I know I've added that mapping, but I want to make sure I test it.